### PR TITLE
Correctly skip test if PMC disabled

### PR DESCRIPTION
**Pull request by choroba**
Fix RT102626.

First, it tries Config::non_bincompat_options to find out, then
Config::Perl::V, and finally checks $Config{ccflags}.

### CODE REVIEW

File: t/parent-pmc.t [hunk 1]
```diff
@@ -14,11 +14,25 @@ use lib 't/lib';
...
-plan skip_all => ".pmc are disabled in this perl"
-    if $no_pmc;
+# Skip this test if perl is compiled with PERL_DISABLE_PMC
+#
+my $pmc = 1;
+if (Config->can('non_bincompat_options')) { # $] ge '5.014'
+    $pmc = 0
+        if grep { $_ eq 'PERL_DISABLE_PMC' } Config::non_bincompat_options();
+} elsif (eval {
+    require Config::Perl::V;
```
Comment (choroba) [inline t/parent-pmc.t:24]: The original patch had

```
require 'Config/Perl/V.pm';
```

Why? Could Config::Perl::V come as `pmc`?

File: t/parent-pmc.t [hunk 2]
```diff
@@ -14,11 +14,25 @@ use lib 't/lib';
 
 plan skip_all => ".pmc are only available with 5.6 and later" if $] < 5.006;
 
-my $no_pmc = defined &Config::non_bincompat_options
```
Comment (choroba) [inline t/parent-pmc.t:17]: `defined` returns false even if `can` returns true (the sub is predeclared in Config.pm).

**Comment by choroba** [comment]
Cc: @haarg @dolmen @wolfsage

**Comment by dolmen** [comment]
I would have appreciated that my authorship of the patch would have been preserved in the commit as my patch was submitted a long time ago in #3.

**Comment by choroba** [comment]
@dolmen Do you mean being mentioned in Changes, or should I have cherry-picked your original commit and start from there? If the latter, I apologise.

**Pull request merged**